lib/purl_listener.rb

```diff
@@ -110,6 +110,8 @@ def listen_for_changes
       listener = Listen.to(path,
                            force_polling: true, # because on NFS mount
                            ignore: [/\.lock$/],
+                           latency: 60,         # wait between polling
+                           wait_for_delay: 5,   # see #157: slow down delta between change notice and action due to NFS
```

Comment:
I've looked into `FileUtils.touch` quite a bit. It uses `File.utime` to change the mtime of the file *if* the file exists (which it doesn't), and so it uses `File.open(fn, 'a') {}` to create the file. Internally to ruby MRI that open/close calls utime_internal evidently. I don't think we can get more atomic than that for Ruby. Using `system("touch fn")` might be better as you're getting actual system calls at the lowest-level. Either way, I don't think we can do that much about the atomic-ness of the creating of the touch file -- I think `FileUtils.touch` is sufficient.

That said, the best thing we can do on the listener side is just slow it down -- `wait_for_delay` inserts a delay between the event (noticing that a file is created) and the callback action, as far as I can tell by digging into the `listen` gem.

As for the 5 second delay -- I dunno what the best value would be. I just chose it because it seems slow enough. 

The `latency` is a 60 second wait between polling -- that's the primary delay here. Again I chose it because it seems slow enough.

In production, we're getting at most a several touch files every minute. So I don't think the slowdown is too much for it. It does add clock time to watching an item go through the pipeline -- i.e., you have to wait a minute before it gets picked up by the listener. But since discovery dispatcher is run so infrequently, I don't think it affects the automated pipeline at all.